src/open_inwoner/accounts/tests/test_oidc_views.py

```diff
@@ -1201,6 +1315,173 @@ def test_new_user_is_created_when_new_kvk(
+
+        self.assertEqual(list(User.objects.all()), [existing_user, new_user])
+
+        self.assertEqual(callback_response.wsgi_request.user, new_user)
+        self.assertTrue(
+            EHerkenningSessionContext(
+                callback_response.wsgi_request
+            ).is_branch_restricted()
+        )
+
+    @patch("open_inwoner.accounts.signals.KvKClient.get_basisprofiel", autospec=True)
+    @patch("open_inwoner.accounts.signals.KvKClient.retrieve_rsin_with_kvk")
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.get_userinfo")
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.store_tokens")
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.verify_token")
+    @patch("mozilla_django_oidc_db.backends.OIDCAuthenticationBackend.get_token")
+    @patch(
+        "open_inwoner.accounts.models.OpenIDEHerkenningConfig.get_solo",
+        return_value=OpenIDEHerkenningConfig(
+            id=1,
+            enabled=True,
+            legal_subject_claim=["sub"],
+            branch_number_claim=["urn:etoegang:1.9:ServiceRestriction:Vestigingsnr"],
+        ),
+    )
+    def test_existing_user_is_returned_for_already_existing_kvk_and_vestiging(
+        self,
+        mock_get_solo,
+        mock_get_token,
+        mock_verify_token,
+        mock_store_tokens,
+        mock_get_userinfo,
+        mock_retrieve_rsin_with_kvk,
+        mock_get_basisprofiel,
+    ):
+        mock_get_basisprofiel.return_value = {
+            "_embedded": {"eigenaar": {"rechtsvorm": "Stichting"}}
+        }
+        mock_retrieve_rsin_with_kvk.return_value = "123456789"
+        # set up a user with a non existing email address
+        mock_get_userinfo.return_value = {
+            "sub": "12345678",
+            "urn:etoegang:1.9:ServiceRestriction:Vestigingsnr": "849586958473",
+        }
+        existing_user = eHerkenningUserFactory.create(
+            kvk="12345678",
+            rsin="123456789",
+            email="existing_user@example.com",
+            vestiging="849586958473",
+        )
+        session = self.client.session
+        session["oidc_states"] = {
+            "mock": {
+                "nonce": "nonce",
+                "config_class": "accounts.OpenIDEHerkenningConfig",
+            }
+        }
+        session.save()
+        callback_url = reverse("eherkenning_oidc:callback")
+
+        self.assertFalse(User.objects.filter(email="new_user@example.com").exists())
+
+        # enter the login flow
+        callback_response = self.client.get(
+            callback_url, {"code": "mock", "state": "mock"}
+        )
+
+        self.assertRedirects(
+            callback_response, reverse("pages-root"), fetch_redirect_response=False
+        )
+        db_user = User.objects.get()
+
+        mock_retrieve_rsin_with_kvk.assert_not_called()
+        self.assertEqual(db_user.email, existing_user.email)
+        self.assertEqual(db_user.kvk, "12345678")
```

Comment:
`self.assertEqual(db_user.kvk, existing_user.kvk)`

Similarly for the lines below